### ISSUE TITLE
Partition appears wrong on kubectl describe

### ISSUE THREAD

**Issue by behos** [kubernetes/kubernetes#90758]
**What happened**:

When using `kubectl describe` on a statefulset, the Partition number shows up as a very high integer:
```
Partition:        824642334680
```
Whereas the real value is 3 as shown in `kubectl get`

```
partition: 3
```

**What you expected to happen**:

I expected the value in `kubectl describe` to be 3.

**How to reproduce it (as minimally and precisely as possible)**:

* Create a stateful set with a partition set
```
  updateStrategy:
    rollingUpdate:
      partition: 3
    type: RollingUpdate
```
* `kubectl describe` the sts

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): Tested on 1.10 and 1.14
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

**Comment by zhouya0** [comment]
Fixed in this PR, update `kubectl` with v1.18 will fix this .https://github.com/kubernetes/kubernetes/pull/85846

**Comment by liggitt** [comment]
thanks for the report, fixed by https://github.com/kubernetes/kubernetes/pull/85846

/close
